### PR TITLE
108: Clicking add dot tool should "forget" selection

### PR DESCRIPTION
**Pull request by rmpowell77**
Fixing issue by adding a boolean on if we support selection for a tool.

## Pre-PR checklist

- [x] Ran `npm run serve` and:
  - [x] Checked basic functionality
  - [x] Checked that errors are handled
- [x] Ran `npm run lint`
- [x] Ran `npm run test:unit`
- [x] Ran `npm run test:e2e` and ran relevant tests
- [x] Attached reviewers to PR and pinged on Slack/email

## Screenshots/GIFs

[Attach screenshots if making a visible change!]

### CODE REVIEW

**Comment by rmpowell77** [comment]
This should address issue #118 .

File: src/components/menu-bottom/MenuBottom.vue [hunk 1]
```diff
@@ -74,6 +74,9 @@ export default Vue.extend({
       ].tool;
       const tool: BaseTool = new ToolConstructor();
       this.$store.commit("setToolSelected", tool);
+      if (!tool.supportsSelection()) {
```
Comment (kmdurand) [inline src/components/menu-bottom/MenuBottom.vue]: Ah nice I like the way you went about this

**Review by kmdurand — APPROVED** [review]

File: src/tools/BaseTool.ts [hunk 2]
```diff
@@ -83,6 +83,9 @@ export default abstract class BaseTool {
   abstract onMouseDown(event: MouseEvent): void;
   abstract onMouseUp(event: MouseEvent): void;
   abstract onMouseMove(event: MouseEvent): void;
+
+  // does the tool support selections
+  abstract supportsSelection(): boolean;
```
Comment (adhamrait) [inline src/tools/BaseTool.ts]: Would it be simpler for this to be not a method? `abstract supportsSelection: boolean;` instead as an abstract property

**Review by adhamrait — APPROVED** [review]

**Review by MichaelTamaki — APPROVED** [review]
Nice work! :rocket:

**Pull request merged**